### DISCUSSION TITLE
How to use image_path features with SageMaker endpoint?

### DISCUSSION THREAD

**Discussion by ijmiller2** [autogluon/autogluon#1659] [Q&A] [answered]
I'm hoping to create a SageMaker endpoint for a multi-modal model I've trained with AutoGluon, but I'm unsure of how to use image paths in this context / environment. Would s3 URIs work? Any tips or suggestions would be much appreciated. An example of or suggestions on how to deploy an ImagePredictor-based model to such an endpoint would be helpful too.

Note that I have tried a few things and posted my results efforts in a related issue a little while back #1644.

**Comment by Innixma** [comment, marked as answer]
Thanks for notifying us of this problem. I'm uncertain if we claim to support image in v0.4 with the SageMaker endpoint logic. The best bet would be to specify the S3 urls as the paths for the image urls, but I don't know if AG can load images from S3 URL at present (if not we should definitely add it though). I've notified the dev team and hopefully this will be fixed in v0.5 or v0.4.1.

**Reply by ijmiller2** [reply]
Great, thank you @Innixma! My best guess is that some logic would need to be added / adapted for the `tabular_serve.py` script used for inference (and used in the example in the docs). But I couldn't quite figure out how to adapt the TabularPredictor class for ImagePredictor or the Multi-Modal class in this context.

**Reply by Innixma** [reply]
Yes, its a bit tricky because you'd need to download the images to disk on the endpoint machine while having the image paths in your data match the new on-disk location on the endpoint. It is probably do-able but likely a fair amount of work. We will be looking into it though as it is important.

**Reply by ijmiller2** [reply]
Right. And I'm not super familiar with how to appropriately handle disk operations on a SageMaker endpoint machine, so I figured that could be a rabbit hole. Supporting S3 URIs would seem like a robust solution that wouldn't bog down or complicate the endpoint's use of disk, but I wasn't sure what that would take in terms of modifying the source code.

Anyway, I appreciate the feedback and support. Looking forward to taking these multi-modal models live for realtime inference!
Ian

**Comment by ijmiller2** [comment]
Hey @Innixma , hope everything has been going well for your and the autogluon team recently! Just wanted to check in to see the status of #1644. Would be great to use the NLP and image classification / multi-modal abilities of AutoGluon within a sagemaker endpoint for a project I'm working on at the moment. Thank you! - Ian
